app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt

```diff
@@ -66,7 +66,9 @@ constructor(
             .post("code=$authenticationCode".toRequestBody())
             .build()
 
-    val responseJson = httpClient.newCall(request).execute().body?.string() ?: return false
+    val responseJson =
+        withContext(Dispatchers.IO) { httpClient.newCall(request).execute().body?.string() }
```

Comment:
I think we can inject dispatchers according to this
https://developer.android.com/kotlin/coroutines/coroutines-best-practices#inject-dispatchers